packages/web-frontend/src/sagas.js

```diff
@@ -606,14 +606,16 @@ function* fetchViewData(parameters, errorHandler) {
   const requestResourceUrl = `view?${queryString.stringify(urlParameters)}`;
 
   try {
-    const viewData = yield call(request, requestResourceUrl, errorHandler);
-    return viewData;
+    return yield call(request, requestResourceUrl, errorHandler);
   } catch (error) {
-    if (error.errorFunction) {
-      yield put(error.errorFunction(error, infoViewKey));
-    } else {
-      console.log(`Failed to handle error: ${error.message}`);
+    let errorMessage = error.message;
+
+    if (error.response) {
+      const json = yield error.response.json();
+      errorMessage = json.error;
     }
+
+    yield put(errorHandler(errorMessage, infoViewKey));
```

Comment:
Always use the designated error handler for the saga. Otherwise another error handler from the saga could handle the error.

Reply:
I'm not a fan of ignoring errorFunctions here. errorFunctions should be checked first and handled globally. E.g. if the user is not logged in, that should be a single UI element saying you're logged out, please log in again, instead of each component on the screen having an error message saying "please log in". In that case, a simple "something went wrong" message on each component, with maybe a retry button would be fine.

It's only if the global error function is not hit that we can assume this is a user error, e.g. wrong dates, and the container that made the request can display the specific error.

```
if (error.errorFunction) {	   
  // global error (e.g. not logged in), handle globally, plus display simple error message in each component
  yield put(error.errorFunction(error, infoViewKey));	
  if (errorHandler) {
    errorHandler('Failed to load data');
  }
} else {
  // user error e.g. bad dates
  if (errorHandler) {
    const json = ...
    ...
    errorHandler(errorMessage, infoViewKey);
  }
}

```

With 500 errors:
Sometimes the server falls over and gives a 500. Usually users cannot do anything about this e.g. server is offline, or there is a bug in the code, apart from try again. 

Generally speaking, an API that returns 500 should not return any information about the error. The server fell over, and it shouldn't have. Instead it should be logged and devs alerted and go fix the error. 

It seems from the screenshot before ("Internal server error: Start period must be equal to or earlier than the end period") that some of our APIs are returning 500 with user errors, this is a problem... these should be 400 Bad Request. If that's the case then that would be a separate ticket to change these.